test/support/slack_client.ex

```diff
@@ -173,49 +130,39 @@ defmodule Cog.Test.Support.SlackClient do
     {:noreply, state}
   end
   def handle_info({{ref, sender}=caller, {:chat_and_wait, room, message, reply_from, nil}}, state) do
-    :timer.sleep(@api_wait_interval)
+    api_wait()
     result = Slack.Web.Chat.post_message(room, %{as_user: true, text: message, token: state.token, parse: :full})
     case result["ok"] do
       false ->
         send(sender, {ref, {:error, result["error"]}})
         {:noreply, state}
       true ->
-        user_id = find_user!(reply_from, state)
+        user_id = find_user_id!(reply_from, state)
         SlackClientState.add_waiter(user_id, caller)
         {:noreply, state}
     end
   end
   def handle_info({{ref, sender}=caller, {:chat_and_wait, room, message, reply_from, edited}}, state) do
-    :timer.sleep(@api_wait_interval)
+    api_wait()
     result = Slack.Web.Chat.post_message(room, %{as_user: true, text: message, token: state.token})
     case result["ok"] do
       false ->
         send(sender, {ref, {:error, result["error"]}})
         {:noreply, state}
       true ->
+        api_wait()
```

Comment:
Maybe? I'm not sure if it's worth it at this stage, though. As it is, it's already a bandaid on the Slack library anyway. 

Is there additional functionality you're envisioning for an MFA-accepting or HOF implementation of `api_wait` that we don't currently have and would benefit from?